providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py

```diff
@@ -32,28 +38,50 @@ def fractional(data: dict, *args: JsonLogicArg) -> typing.Optional[str]:
         return None
 
     hash_ratio = abs(mmh3.hash(bucket_by)) / (2**31 - 1)
-    bucket = int(hash_ratio * 100)
+    bucket = hash_ratio * 100
 
+    total_weight = 0
+    fractions = []
     for arg in args:
-        if (
-            not isinstance(arg, (tuple, list))
-            or len(arg) != 2
-            or not isinstance(arg[0], str)
-            or not isinstance(arg[1], int)
-        ):
-            logger.error("Fractional variant weights must be (str, int) tuple")
-            return None
-    variant_weights: typing.Tuple[typing.Tuple[str, int]] = args  # type: ignore[assignment]
-
-    range_end = 0
-    for variant, weight in variant_weights:
-        range_end += weight
+        fraction = __parse_fraction(arg)
+        if fraction:
+            fractions.append(fraction)
+            total_weight += fraction.weight
```

Comment:
In both go & java, an error in parsing makes the full evaluation invalid. To be consistent, I think we should handle the parsing errors and propagate them.

Reply:
i will extract this into an own bug, as this is definitely worth an own pr and an own merge commit

Reply:
extracted into #92

Reply:
Thank you @aepfli